src/js/index.js

```diff
@@ -56,38 +59,90 @@ RequestUtil.json = function (options = {}) {
         </Provider>),
         domElement);
     } else {
-      setTimeout(function () {
-        let routes = RouterUtil.buildRoutes(appRoutes.getRoutes());
-        let router = Router.run(routes, function (Handler, state) {
-          Config.setOverrides(state.query);
-          ReactDOM.render(
-            (<Provider store={PluginSDK.Store}>
-              <Handler state={state} />
-            </Provider>),
-            domElement);
+      let loadedResources = {
+        mesos: false
+      };
+      let mesosEvents = [MESOS_SUMMARY_CHANGE, MESOS_SUMMARY_REQUEST_ERROR];
+
+      // Responsible for checking if application is ready to render
+      function delayOrLoadApplication() {
+        // Check if we've loaded all resources first
```

Comment:
I can't think of a scenario where `renderApplicationToDOM` would get called twice. Because it doesn't go to that portion of the logic until everything has been loaded and at that point we're just waiting for the delay to pass.